egs/discophone/v1/local/chain/tuning/run_tdnn_1b.sh

```diff
@@ -0,0 +1,279 @@
+        else
+          echo "$0: data/lang_chain/$lang_name already exists and seems to be older than data/lang/$lang_name ..."
+          echo " ... not sure what to do.  Exiting."
+          exit 1;
+        fi
+      else
+        mkdir -p data/lang_chain
+        cp -r $langdir data/lang_chain/$lang_name
+        silphonelist=$(cat data/lang_chain/$lang_name/phones/silence.csl) || exit 1;
+        nonsilphonelist=$(cat data/lang_chain/$lang_name/phones/nonsilence.csl) || exit 1;
+        # Use our special topology... note that later on may have to tune this
+        # topology.
+        steps/nnet3/chain/gen_topo.py $nonsilphonelist $silphonelist >data/lang_chain/$lang_name/topo
+      fi
+    fi
+    
+    if [ $stage -le 15 ] && [ $stop_stage -gt 15 ] ; then
+      # Get the alignments as lattices (gives the chain training more freedom).
+      # use the same num-jobs as the alignments
+      steps/align_fmllr_lats.sh --nj $nj_align_fmllr_lats --cmd "$train_cmd" ${lores_train_data_dir} \
+        $langdir $gmm_dir $lat_dir
+      rm $lat_dir/fsts.*.gz # save space
+    fi
+    
+    if [ $stage -le 16 ] && [ $stop_stage -gt 16 ] ; then
+      # Build a tree using our new topology.  We know we have alignments for the
+      # speed-perturbed data (local/nnet3/run_ivector_common.sh made them), so use
+      # those.
+      if [ -f $tree_dir/final.mdl ]; then
+        echo "$0: $tree_dir/final.mdl already exists, refusing to overwrite it."
+        exit 1;
+      fi
+      steps/nnet3/chain/build_tree.sh --frame-subsampling-factor 3 \
+          --context-opts "--context-width=2 --central-position=1" \
+          --leftmost-questions-truncate -1 \
+          --cmd "$train_cmd" 4000 ${lores_train_data_dir} data/lang_chain/$lang_name $ali_dir $tree_dir
+    fi
+    
+    xent_regularize=0.1
+
+
+    if [ $stage -le 17 ] && [ $stop_stage -gt 17 ] ; then
+      echo "$0: creating neural net configs using the xconfig parser";
+      feat_dim=$(feat-to-dim scp:${train_data_dir}/feats.scp -)
+      num_targets=$(tree-info $tree_dir/tree | grep num-pdfs | awk '{print $2}')
+      learning_rate_factor=$(echo "print (0.5/$xent_regularize)" | python)
+      opts="l2-regularize=0.002"
+      linear_opts="orthonormal-constraint=1.0"
+      output_opts="l2-regularize=0.0005 bottleneck-dim=256"
+    
+      mkdir -p $dir/configs
+      cat <<EOF > $dir/configs/network.xconfig
+      input dim=100 name=ivector
+      input dim=$feat_dim name=input
+      # please note that it is important to have input layer with the name=input
+      # as the layer immediately preceding the fixed-affine-layer to enable
+      # the use of short notation for the descriptor
+      fixed-affine-layer name=lda input=Append(-1,0,1,ReplaceIndex(ivector, t, 0)) affine-transform-file=$dir/configs/lda.mat
+      # the first splicing is moved before the lda layer, so no splicing here
+      relu-batchnorm-dropout-layer name=tdnn1 $opts dim=1280
```

Comment:
Can we use the `tdnnf-layer` component here instead? I think it's not the same after all, tdnnf learns a SVD-like weight matrix factorization during training, sorry for the confusion...

Reply:
> Can we use the `tdnnf-layer` component here instead? I think it's not the same after all, tdnnf learns a SVD-like weight matrix factorization during training, sorry for the confusion...

Yes we can. I will have to go through a test with tdnnf-layer instead of the one mentioned above. I will create a different file e.g. named run_tdnn_1c.sh. I will keep layers in run_tdnn_1b.sh unchanged.

Reply:
Update: I have created a new file run_tdnn_1g.sh which uses tdnnf-layers as neural network architecture. That script is located in the same directory as run_tdnn.sh and run_tdnn_1b.sh.
This script has passed test on all babel and globalphone languages.